saul-core/doc/MODELS.md

```diff
@@ -1,6 +1,101 @@
+but the predictions are made in way that `constraintExpression` is hold. There is no limitation for the type of local classifiers.
+They can be SVMs, decision trees or any other learning models available in Saul, [here](https://github.com/IllinoisCogComp/lbjava/blob/master/lbjava/doc/ALGORITHMS.md)
+                                                                                  and [here](https://github.com/IllinoisCogComp/saul/blob/master/saul-core/src/main/java/edu/illinois/cs/cogcomp/saul/learn/SaulWekaWrapper.md).
+
+<a name="IBT">
+##Inference-based Learning.
+
+For the inference based models the basic definitions are exactly similar to the L+I models. In other words, the programmer
+just needs to define the `Learnables` and `ConstrainedClassifiers`. However, to train the ConstrainedClassifiers jointly, instead of
+training local classifiers independently, there are a couple of joint training functions that can be called.
+These functions receive the list of constrained classifiers as input and train their parameters jointly. In contrast to
+L+I models here the local classifiers can be defined as `SparsePerceptrons` or `SparseNetworks` only. This is because the
+joint training should have its own strategy for the wight updates of the involved variables (those variables come down to be the outputs of the local classifiers here).
+For the two cases the programmer can use
+
+```scala
+ JointTrainSparseNetwork.train(param1,param2,...) /* a list of parameters go here*/
+ JointTrainSparsePerceptron.train(param1,param2,...) /*a list of parameters here*/
+```
+
+ JointTrainSparseNetwork.train(badge, cls, 5, init = true, lossAugmented = true)
+
+The list of parameters are the following:
+
+-param1: The name of a global node in the data model that itself or the connected nodes to it are used by the involved `Learnable`s.
+
+-param2: The list of ConstainedClassifiers
+
+-param3: The number of iterations over the training data.
+
+-param4: If the local classifiers should be cleaned from the possibly pre-trained values and initialized by zero weights, this parameter should be true.
+
+-param5: If the approach uses the loss augmented objective for making inference, see below for description.
+
+###Basic approach
+
+The Basic approach for training the models jointly is to do a global prediction at each step of the training and if the
+predictions are wrong update the weights of the related variables.
+
+###Loss augmented
+
+The loss-augmented approach adds the loss of the prediction explicitly to the objective of the training and finds the most violated output;
+it updates the weights of the model according to the errors made in the most violated output.
+This is an approach that is used in structured SVMs and Structured Perceptrons. However, considering an arbitrary loss in the objective
+will make complexities in the optimization, therefore in the implemented version, here, we assume the loss is decomposed similar to
```

Comment:
I think, I meant to say in general "complexities in the optimization" not only computational intractability. Sometimes, it can be even tractable but not straightforward to solve using the same objective's structure.